File: software/examples/vbnet/ExampleWaterBoiling.vb
Imports System
Imports Tinkerforge

Module ExampleWaterBoiling
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Temperature IR Bricklet

    ' Callback subroutine for object temperature reached callback (parameter has unit °C/10)
    Sub ObjectTemperatureReachedCB(ByVal sender As BrickletTemperatureIR, _
                                   ByVal temperature As Short)
        Console.WriteLine("Object Temperature: " + (temperature/10.0).ToString() + " °C")
        Console.WriteLine("The water is boiling!")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim tir As New BrickletTemperatureIR(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Set emissivity to 0.98 (emissivity of water, 65535 * 0.98 = 64224.299)
        tir.SetEmissivity(64224)

        ' Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        tir.SetDebouncePeriod(10000)

        ' Register object temperature reached callback to subroutine ObjectTemperatureReachedCB
        AddHandler tir.ObjectTemperatureReached, AddressOf ObjectTemperatureReachedCB

        ' Configure threshold for object_temperature "greater than 100 °C" (unit is °C/10)
        tir.SetObjectTemperatureCallbackThreshold(">"C, 100*10, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/javascript/ExampleWaterBoiling.js
var IPConnection = require('Tinkerforge/IPConnection');
var BrickletTemperatureIR = require('Tinkerforge/BrickletTemperatureIR');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'kqJ';// Change to your UID

var ipcon = new IPConnection();// Create IP connection
var tir = new BrickletTemperatureIR(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set emissivity to 0.98 (emissivity of water)
        tir.setEmissivity(parseInt(0xFFFF*0.98));
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        tir.setDebouncePeriod(10000);
        // Configure threshold for "greater than 100 °C" (unit is °C/10)
        tir.setObjectTemperatureCallbackThreshold('>', 100*10, 0);    
    }
);

//Register threshold reached callback
tir.on(BrickletTemperatureIR.CALLBACK_OBJECT_TEMPERATURE_REACHED,
    //Callback for object temperature greater than 100 °C
    //(parameter has unit °C/10)
    function(temp) {
        console.log('The surface has a temperature of ' + temp/10.0 + ' °C.');
        console.log('The water is boiling!');
        console.log();
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

